driver-sync/src/main/com/mongodb/client/internal/CryptConnection.java

```diff
@@ -0,0 +1,225 @@
+            throw new MongoClientException("Auto-encryption requires a minimum MongoDB version of 4.2");
+        }
+
+        BasicOutputBuffer bsonOutput = new BasicOutputBuffer();
+        BsonBinaryWriter bsonBinaryWriter = new BsonBinaryWriter(createBsonWriterSettings(), createBsonBinaryWriterSettings(),
+                bsonOutput, getFieldNameValidator(payload, commandFieldNameValidator, payloadFieldNameValidator));
+        BsonWriter writer = payload == null
+                ? bsonBinaryWriter
+                : new SplittablePayloadBsonWriter(bsonBinaryWriter, bsonOutput, createSplittablePayloadMessageSettings(), payload);
+
+        getEncoder(command).encode(writer, command, EncoderContext.builder().build());
+
+        RawBsonDocument encryptedCommand = crypt.encrypt(database,
+                new RawBsonDocument(bsonOutput.getInternalBuffer(), 0, bsonOutput.getSize()));
+
+        RawBsonDocument encryptedResponse = wrapped.command(database, encryptedCommand, commandFieldNameValidator, readPreference,
+                new RawBsonDocumentCodec(), sessionContext, responseExpected, null, null);
+
+        RawBsonDocument decryptedResponse = crypt.decrypt(encryptedResponse);
+
+        BsonBinaryReader reader = new BsonBinaryReader(decryptedResponse.getByteBuffer().asNIO());
+
+        return commandResultDecoder.decode(reader, DecoderContext.builder().build());
+    }
+
+    @Override
+    public <T> T command(final String database, final BsonDocument command, final FieldNameValidator fieldNameValidator,
+                         final ReadPreference readPreference, final Decoder<T> commandResultDecoder, final SessionContext sessionContext) {
+        return command(database, command, fieldNameValidator, readPreference, commandResultDecoder, sessionContext, true, null, null);
+    }
+
+    @SuppressWarnings("unchecked")
+    private Codec<BsonDocument> getEncoder(final BsonDocument command) {
+        return (Codec<BsonDocument>) REGISTRY.get(command.getClass());
+    }
+
+    private FieldNameValidator getFieldNameValidator(final SplittablePayload payload,
+                                                     final FieldNameValidator commandFieldNameValidator,
+                                                     final FieldNameValidator payloadFieldNameValidator) {
+        if (payload == null) {
+            return commandFieldNameValidator;
+        }
+
+        Map<String, FieldNameValidator> rootMap = new HashMap<String, FieldNameValidator>();
+        rootMap.put(payload.getPayloadName(), payloadFieldNameValidator);
+        return new MappedFieldNameValidator(commandFieldNameValidator, rootMap);
+    }
+
+    private BsonWriterSettings createBsonWriterSettings() {
+        return new BsonWriterSettings();
+    }
+
+    // TODO
+    // Currently these settings will allow a split point that is potentially too large after encryption occurs. Need to find a way
```

Comment:
I don't recall but does the `SplittablePayloadMessage` have a min size of 1 anyway?

Reply:
I'm talking about size in bytes here, not size in documents.  Is that what you mean?

Reply:
I guess I'm not 100% sure on the todo comment